js/communication/geppettoJupyter/PythonControlledCapability.js

```diff
@@ -13,15 +13,26 @@ define(function (require) {
 
   module.exports = {
     createPythonControlledComponent (WrappedComponent) {
+      if (typeof WrappedComponent !== "function") {
+        // Fixes components defined as objects (e.g. Material-ui components)
+        class Wrapper extends React.Component {
+          render () {
+            return <WrappedComponent {...this.props} />;
+          }
+        } 
+        WrappedComponent = Wrapper;
+      }
+      
       class PythonControlledComponent extends WrappedComponent {
         constructor (props) {
           super(props);
           if (this.state == undefined) {
             this.state = {};
           }
           this.state.model = props.model;
-          this.state.componentType = WrappedComponent.name;
+          this.state.componentType = WrappedComponent.name || WrappedComponent.Naked.render.name;
```

Comment:
this can be wrapped by a getter in case the HOC wrapper needs to get base component from different libraries, just a suggestion, no need to change it now but I will place a comment in the code for the future.

Reply:
Where do you suggest to have the getter? The `PythonControlledComponent` class?

Reply:
yes